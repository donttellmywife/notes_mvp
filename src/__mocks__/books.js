const books = [{
    id: 1,
    userId: 1,
    name: 'Program',
    'public': true,
    user: {
        id: 1,
        avatar: 'https://a.ppy.sh/369435_1325681849.jpg',
        name: 'Cthulhu Fhtagn',
        address: {
            country: 'Pacific Ocean',
            city: 'R’lyeh'
        },
        login: 'cthulhu'
    }
}, {
    id: 3,
    userId: 1,
    name: 'calculate Paradigm',
    'public': true,
    user: {
        id: 1,
        avatar: 'https://a.ppy.sh/369435_1325681849.jpg',
        name: 'Cthulhu Fhtagn',
        address: {
            country: 'Pacific Ocean',
            city: 'R’lyeh'
        },
        login: 'cthulhu'
    }
}, {
    userId: 1,
    'public': true,
    name: 'new book',
    id: 64
}, {
    userId: 1,
    'public': true,
    name: 'qwe',
    id: 65
}]

export default books
