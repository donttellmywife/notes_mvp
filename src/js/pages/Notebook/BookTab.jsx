import React from 'react'
import { string, func, bool, array, number } from 'prop-types'
import { connect } from 'react-redux'
import { propEq, find, compose } from 'ramda'

// action creators
import { bookSelected, deleteBook, updateBook } from './redux/ac'

// styles
import { active, bookTab } from './styles/tab.sass'


function BookTab(props) {
  const { name, select, remove, update, isSelected } = props
  const activeCls = isSelected ? active : ''

  return (<div
    onClick={select}
    className={`${bookTab} ${activeCls}`}>
      <h1>{name}</h1>
      <input defaultValue={name} />
      <button onClick={remove}>Delete</button> |
      <button onClick={update}> Edit</button>
  </div>)
}

BookTab.propTypes = {
  id: number,
  title: string,
  name: string,
  isSelected: bool
}
BookTab.defaultProps = {
  id: -1,
  title: 'Book',
  name: 'Some name',
  isSelected: false,
}

const mapStateToProps = (state, ownProps) => {
  const {items, selectedId} = state.books
  const {id, name} = find(propEq('id', ownProps.id), items)
  return {
    id,
    name,
    isSelected: id === selectedId,
  }
}
const mapDispatchToProps = (dispatch) => ({
  select: compose(dispatch, bookSelected),
  remove: compose(dispatch, deleteBook),
  update: compose(dispatch, updateBook),
})
const mergeProps = (stateProps, dispatchProps) => {
  const { id, name, isSelected } = stateProps
  const { select, remove, update } = dispatchProps
  return {
    name,
    isSelected,
    select: () => select(id),
    remove: () => remove(id),
    update: () => {
      const {name} = stateProps
      update({id, name})
    },
  }
}

export const Unwrapped = BookTab
export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(BookTab)
