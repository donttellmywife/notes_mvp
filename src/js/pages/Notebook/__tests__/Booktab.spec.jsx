import React from 'react'
import Enzyme, { shallow }  from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

import BookTab, { Unwrapped } from '../BookTab'

import books from '../../../../__mocks__/books.js'


describe('Booktab component: ', () => {
  const book = books[0]

  it('Booktab renders correctrly', () => {
    const component = shallow(<Unwrapped {...book} />)
    expect(component).toMatchSnapshot()
  })
})
