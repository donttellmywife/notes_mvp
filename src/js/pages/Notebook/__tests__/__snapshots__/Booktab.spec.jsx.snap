// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booktab component:  Booktab renders correctrly 1`] = `
<div
  className="undefined "
>
  <h1>
    Program
  </h1>
  <input
    defaultValue="Program"
  />
  <button>
    Delete
  </button>
   |
  <button>
     Edit
  </button>
</div>
`;
